cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Transaction.hs

```diff
@@ -115,11 +115,16 @@ hprop_transaction = integrationRetryWorkspace 0 "simple transaction build" $ \te
     , "--out-file", txbodySignedFp
     ]
 
-  void $ execCli' execConfig
+  stdout <- execCli' execConfig
     [ anyEraToString cEra, "transaction", "submit"
     , "--tx-file", txbodySignedFp
     ]
 
+  txId <- if null stdout
+          then do
+            H.note_ "Expected output of \"transaction submit\" not to be empty"
+            H.failure
+          else pure $ last $ lines stdout
```

Comment:
This is a poor way of obtaining the txid The txid can be calculated from the tx itself and you should query the mempool to confirm indeed if it was submitted or not.